cli/internal/kubernetes/kubernetes.go

```diff
@@ -10,3 +10,77 @@ Package kubernetes provides functions to interact with a live cluster to the CLI
 Currently it is used to implement the status and upgrade commands.
 */
 package kubernetes
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func newClient() (kubernetes.Interface, error) {
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", constants.AdminConfFilename)
+	if err != nil {
+		return nil, fmt.Errorf("building kubernetes config: %w", err)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("setting up kubernetes client: %w", err)
+	}
+	return kubeClient, nil
+}
+
+// StableInterface is an interface to interact with stable resources.
+type StableInterface interface {
+	GetCurrentConfigMap(ctx context.Context, name string) (*corev1.ConfigMap, error)
+	UpdateConfigMap(ctx context.Context, configMap *corev1.ConfigMap) (*corev1.ConfigMap, error)
+	CreateConfigMap(ctx context.Context, configMap *corev1.ConfigMap) (*corev1.ConfigMap, error)
+	KubernetesVersion() (string, error)
+}
+
+// NewStableClient returns a new StableClient.
+func NewStableClient() (StableInterface, error) {
+	client, err := newClient()
+	if err != nil {
+		return nil, err
+	}
+	return NewStableClientWithK8sClient(client), nil
+}
+
+// NewStableClientWithK8sClient returns a new StableInterface.
+func NewStableClientWithK8sClient(client kubernetes.Interface) StableInterface {
+	return &stableClient{client: client}
+}
```

Comment:
Why do we have 2 functions for this now?
The first one is called in `runUpgradeApply`, the second one in `runStatus`.

IMO, we should not use the first function, as it assumes filepaths outside of the `cmd` package, something I tried to fix in https://github.com/edgelesssys/constellation/pull/2148

Ideally, we only have one function named `NewStableClient`, which takes either a `kubernetes.Interface`, or the path to a kubeconfig file as argument and returns a `stableClient`.

Reply:
good point, fixed it.